### PR TITLE
feat: Add Parquet as a batch encoding option

### LINKED ISSUE #1148: Feature proposal: Add Parquet as a Batch encoding option
### Feature scope

Taps (catalog, state, stream maps, etc.)

### Description

Currently the only Batch encoding option is JSONL. This feature will add support for writing batch files as parquet.

### PR DESCRIPTION
**Pull request by jamielxcarter**
Closes #1148

----
:books: Documentation preview :books:: https://meltano-sdk--1235.org.readthedocs.build/en/1235/

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -57,8 +57,13 @@ python-dotenv = ">=0.20,<0.22"
 typing-extensions = "^4.2.0"
 simplejson = "^3.17.6"
 jsonschema = "^4.16.0"
+pyarrow = "^10.0.1"
```
Comment (edgarrmondragon) [inline pyproject.toml:60]: @jamielxcarter I would like to keep `pyarrow` and `numpy` as optional dependencies. Is it possible to add them to an `extra` array?

Comment (jamielxcarter) [reply pyproject.toml:60]: Yeah, I'll change them back to being optional

**Comment by tayloramurphy** [comment]
@edgarrmondragon where are we with this PR? cc @seth-meltano @aaronsteers

**Comment by edgarrmondragon** [comment]
> @edgarrmondragon where are we with this PR? cc @seth-meltano @aaronsteers

@tayloramurphy I have not taken a look since it was still a draft. Though, pyarrow and numpy have some compatibility issues with 3.7 and 3.11 (need to reconcile the two with different dep versions), so I think that's still missing from this PR.

**Comment by tayloramurphy** [comment]
@jamielxcarter are you still willing to help make progress on this PR? Would be great to get it in.

**Comment by tayloramurphy** [comment]
@jamielxcarter I'm going to close this since there's no activity. But this would be great to get in if you're still willing to contribute!

**Pull request closed without merging**